python/tests/test_topology.py

```diff
@@ -45,43 +45,52 @@ def slice(
         record_provenance=True):
     """
     A clearer but slower implementation of TreeSequence.slice() defined in trees.py
+    expanded to slice at multiple intervals
     """
     if start is None:
-        start = 0
+        start = [0]
     if stop is None:
-        stop = ts.sequence_length
-
-    if start < 0 or stop <= start or stop > ts.sequence_length:
+        stop = [ts.sequence_length]
+    zipRanges = list(zip(start, stop))
+    # not sure how best to reset coords if slicing multiple intervals
+    # so currently defaulting to False
+    if len(zipRanges) > 1:
+        reset_coordinates = False
+    if any(x < 0 for x in start) or any(x[1] <= x[0] for x in zipRanges) \
+       or any(x > ts.sequence_length for x in stop):
         raise ValueError("Slice bounds must be within the existing tree sequence")
     tables = ts.dump_tables()
     tables.edges.clear()
     tables.sites.clear()
     tables.mutations.clear()
     for edge in ts.edges():
-        if edge.right <= start or edge.left >= stop:
-            # This edge is outside the sliced area - do not include it
-            continue
-        if reset_coordinates:
-            tables.edges.add_row(
-                max(start, edge.left) - start, min(stop, edge.right) - start,
-                edge.parent, edge.child)
-        else:
-            tables.edges.add_row(
-                max(start, edge.left), min(stop, edge.right),
-                edge.parent, edge.child)
-    for site in ts.sites():
-        if start <= site.position < stop:
+        for aRange in zipRanges:
+            if edge.right <= aRange[0] or edge.left >= aRange[1]:
+                # This edge is outside the sliced area - do not include it
+                continue
             if reset_coordinates:
```

Comment:
will `reset_coordinates` just shift everything to `start[0]` or snip out all between-interval spaces? I vote for the latter.

Reply:
currently I'm not dealing with it at all. I figure that if there is more than one interval we probably don't want to allow a coordinate reset.

Reply:
I'd say why not allow it? It wouldn't be hard to shift the coordinates to totally snip out those bits.